### PR TITLE
[FEATURE] Inherit dependencies from a workspace

### LINKED ISSUE #231: [FEATURE] Inherit dependencies from a workspace
Dependencies can be inherited from a workspace by specifying the dependency in the workspace's [[workspace.dependencies]](https://doc.rust-lang.org/nightly/cargo/reference/workspaces.html#the-dependencies-table) table. After that, add it to the [dependencies] table with workspace = true.

### PR DESCRIPTION
**Pull request by deveusss**
closes #231

### CODE REVIEW

**Review by Buckram123 — APPROVED** [review]
Love it

File: Cargo.toml [hunk 1]
```diff
@@ -19,3 +19,59 @@ debug-assertions = false
...
+cw-rules = { version = "*", path = "./contracts/cw-rules" }
+cw-croncat-core = { version = "0.1.5", path = "./packages/cw-croncat-core" }
+cw721-base = "0.15.0"
+
+[workspace.metadata.scripts]
+optimize = """docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/rust-optimizer:0.12.5
+"""
```
[the remark this answers is not in the document]
Comment (deveusss) [reply Cargo.toml:77]: maybe will, but I am not sure

File: justfile [hunk 2]
```diff
@@ -48,14 +48,14 @@ optimize:
 		--mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
 		--mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
 		--platform linux/amd64 \
-		cosmwasm/workspace-optimizer:0.12.8
+		cosmwasm/workspace-optimizer:0.12.10
```
[the remark this answers is not in the document]
Comment (deveusss) [reply justfile:51]: Yea, new cargo feature works only with this optimizer)

File: .github/workflows/Basic.yml [hunk 3]
```diff
@@ -48,7 +48,7 @@ jobs:
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: 1.63.0
+          toolchain: 1.64.0
```
[the remark this answers is not in the document]
Comment (deveusss) [reply .github/workflows/Basic.yml:51]: without this update, ci fails)

[the remark this answers is not in the document]
Comment (Buckram123) [reply .github/workflows/Basic.yml:51]: @deveusss I mean even higher, 1.66.0?)

[the remark this answers is not in the document]
Comment (deveusss) [reply .github/workflows/Basic.yml:51]: last is 1.65.0))

**Pull request merged**